src/utils/api_types.ts

```diff
@@ -41,15 +41,28 @@ export interface Project {
   envVars: string[];
 }
 
-export type Organization = UserOrganization | NormalOrganization;
+export type Organization =
+  & (UserOrganization | NormalOrganization)
+  & CommonOrganization;
 
 export interface UserOrganization {
-  id: string;
   name: null;
 }
 
 export interface NormalOrganization {
+  name: string;
+}
+
+export interface CommonOrganization {
   id: string;
+  members: OrganizationMember[];
+}
+
+export interface OrganizationMember {
+  user: User;
+}
```

Comment:
I personally find it a bit hard to get what the resulting struct looks like - so if I understand it correctly the result is as follows

```ts
// UserOrganization
{
  id: string;
  name: null;
  members: OrganizationMember[];
}

// NormalOrganization
{
  id: string;
  name: string;
  members: OrganizationMember[];
}
```

What I would do instead is like the following, I feel it's a bit easier to read, plus with this representation `NormalOrganization` and `UserOrganization` have `id` and `members` properties, which better reflects the org data structure

```ts
type CommonOrganization = {
  id: string;
  members: OrganizationMember[];
};

export type NormalOrganization = CommonOrganization & {
  name: string;
};

export type UserOrganization = CommonOrganization & {
  name: null;
};

export type Organization = NormalOrganization | UserOrganization;
```

Reply:
Done!